### PR TITLE
Add Nuget projects.

### PR DESCRIPTION
**Pull request by Ayrx**
This PR adds two EFDA projects for Nuget-based .NET projects.

1. In `efda/csharp/nuget/with-packages.config`, we have the dependencies specified in the `packages.config` file.
2. In `efda/csharp/nuget/with-packagereference`, we have the dependencies specified as `PackageReference` directives in the `.csproj`. file.

The diff is a bit messy right now as it includes the changes in #2. I'll rebase this PR once that has been merged so it's a little bit easier to review.

### CODE REVIEW

File: csharp/nuget/with-packagereference/info.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "vulnerabilities": 1,
+    "direct_dependencies"; 4,
```
Comment (hendychua) [inline csharp/nuget/with-packagereference/info.json]: There's a typo here: `;` -> `:`.

File: csharp/nuget/with-packages.config/info.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "vulnerabilities": 1,
+    "direct_dependencies"; 4,
```
Comment (hendychua) [inline csharp/nuget/with-packages.config/info.json]: There's a typo here: `;` -> `:`.

**Review by hendychua — CHANGES_REQUESTED** [review]
Looks good apart from a couple of typos. Good job finding the vulnerable method. Be prepared to explain the thought and research process of finding the vulnerable method (You must be certain that it is a vulnerable method).

**Comment by Ayrx** [comment]
Rebased and ready for review.

**Review by hendychua — APPROVED** [review]

**Pull request merged**